src/App.js

```diff
@@ -1,82 +1,9 @@
-import { useEffect, useState } from "react";
 import logo from "./assets/img/mlh-prep.png";
-import { toast } from "react-toastify";
-import "react-toastify/dist/ReactToastify.css";
+import useWeather from "./helpers/customHooks/useWeather";
 
 const App = () => {
-  const [error, setError] = useState(null);
-  const [isLoaded, setIsLoaded] = useState(false);
-  const [city, setCity] = useState("");
-  const [results, setResults] = useState(null);
-
-  useEffect(()=>{
-	const options = {
-		enableHighAccuracy: false,
-		timeout: 5000,
-		maximumAge: Infinity,
-	  };
-  
-	  function onSuccess(position) {
-		let latitude = position.coords.latitude;
-		let longitude = position.coords.longitude;
-  
-		fetch(
-		  `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${process.env.REACT_APP_APIKEY}`
-		)
-		  .then((res) => res.json())
-		  .then(
-			(result) => {
-			  setCity(result.name);
-			},
-			(error) => {
-			  setIsLoaded(true);
-			  setError(error);
-			}
-		  );
-	  }
-  
-	  function onError(error) {
-		  toast.configure();
-		  toast.error("Location access denied.", {
-		  theme: "light",
-		  hideProgressBar: false,
-		  closeButton: true,
-		  position: "top-right",
-		  autoClose: 3000,
-		});
-		setCity("New York City");
-	  }
-  
-	  window.navigator.geolocation.getCurrentPosition(onSuccess, onError, options);
-	  
-  },[])
-
-  useEffect(() => {
-    if (city !== "") {
-      fetch(
-        "https://api.openweathermap.org/data/2.5/weather?q=" +
-          city +
-          "&units=metric" +
-          "&appid=" +
-          process.env.REACT_APP_APIKEY
-      )
-        .then((res) => res.json())
-        .then(
-          (result) => {
-            if (result["cod"] !== 200) {
-              setIsLoaded(false);
-            } else {
-              setIsLoaded(true);
-              setResults(result);
-            }
-          },
-          (error) => {
-            setIsLoaded(true);
-            setError(error);
-          }
-        );
-    }
-  }, [city]);
+  const { city, results, isLoaded, setCity, setResults, setIsLoaded, error } =
```

Comment:
I don't think we will be using setResults anywhere because it is just setting the api response. Right? And we are doing all the api requests in useWeather hook only.  Will it be alright if we remove it from here?

Reply:
Oh right! I'll remove setResults